GetPMWallet/handler.ts

```diff
@@ -44,37 +78,143 @@ type IHttpHandler = (
+              {
+                brand: info.brand,
+                expireMonth: info.expireMonth,
+                expireYear: info.expireYear,
+                holder: info.holder,
+                hpan: info.hashPan,
+                masked_pan: info.blurredNumber,
+                type: CardTypeEnum.Card
+              } as PublicCreditCard
+            ];
+          }
+          if (
+            wallet.walletType === WalletTypeEnum.Bancomat &&
+            WalletCardInfoInput.is(info)
+          ) {
+            return [
+              ...acc,
+              {
+                brand: info.brand,
+                expireMonth: info.expireMonth,
+                expireYear: info.expireYear,
+                holder: info.holder,
+                hpan: info.hashPan,
+                masked_pan: info.blurredNumber,
+                type: CardTypeEnum.Bancomat
+              } as PublicCreditCard
+            ];
+          }
+          if (
+            wallet.walletType === WalletTypeEnum.BPay &&
+            WalletBpayInfoInput.is(info)
+          ) {
+            return [
+              ...acc,
+              {
+                bank_name: info.bankName,
+                hpan: info.uidHash,
+                type: BPayTypeEnum.BPay
+              } as PublicBPay
+            ];
+          }
+          if (
+            wallet.walletType === WalletTypeEnum.Satispay &&
+            WalletSatispayInfoInput.is(info)
+          ) {
+            return [
+              ...acc,
+              {
+                hpan: info.uuid,
+                type: TypeEnum.Satispay
+              } as PublicSatispay
+            ];
+          }
+          return acc;
+        }, [] as ReadonlyArray<PublicWalletItem>)
+      )
+      .getOrElse([])
+  });
+};
+
 export function GetPMWalletHandler(
-  _: TypeofApiCall<GetWalletV2T>
+  getWalletV2: TypeofApiCall<GetWalletV2T>,
+  subscriptionKey: NonEmptyString
 ): IHttpHandler {
-  return async (context, __) => {
-    // TODO: Missing implementation, the method is mocked.
-    const mockWallet1: PublicCreditCard = {
-      brand: "VISA",
-      expireMonth: "1",
-      expireYear: "2023",
-      holder: "Mario Bianchi",
-      hpan: "807ae5f38db47bff8b09b37ad803cb10ef5147567a89a33a66bb3282df4ad966",
-      masked_pan: "4763",
-      type: TypeEnum.Card
-    };
-    const mockWallet2: PublicCreditCard = {
-      brand: "MASTERCARD",
-      expireMonth: "7",
-      expireYear: "2027",
-      holder: "Mario Bianchi",
-      hpan: "7726b99f6eff4f80f27e91eee2fb4f6e9f7aa01c5837cbc9f1b9dc4c51689a29",
-      masked_pan: "7778",
-      type: TypeEnum.Bancomat
-    };
-    return taskEither
-      .of(ResponseSuccessJson({ data: [mockWallet1, mockWallet2] }))
-      .fold<IResponseSuccessJson<Wallet> | ErrorTypes>(err => {
+  return async (context, { fiscalCode }) => {
+    return tryCatch(
+      () =>
+        getWalletV2({
+          "Fiscal-Code": fiscalCode,
+          SubscriptionKey: subscriptionKey
```

Comment:
Why don't we encapsulate `SubscriptionKey` into a `withDefaults` in client definition?

Reply:
Fixed in 95a58e5, I've done some refractory on how the Client is created